### PR TITLE
EventGrid: Added title and description; Added specs/examples for new API version.

### PR DESCRIPTION
**Pull request by kalyanaj**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by kalyanaj** [comment]
@jhendrixMSFT, per your earlier comments, I have created this PR now using the master branch. Please let me know if you have any other feedback.

**Comment by kalyanaj** [comment]
FYI, the one linter error reported is due to a known validator issue being tracked by https://github.com/Azure/azure-openapi-validator/issues/109.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json [hunk 1]
```diff
@@ -0,0 +1,2055 @@
...
+            }
+        },
+        "/subscriptions/{subscriptionId}/providers/Microsoft.EventGrid/topics": {
+            "get": {
+                "tags": [
+                    "Topics"
+                ],
+                "summary": "List topics under an Azure subscription",
+                "description": "List all the topics under an Azure subscription",
+                "operationId": "Topics_ListBySubscription",
```
Comment (anuchandy) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json]: `Topics_List()`

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json [hunk 2]
```diff
@@ -0,0 +1,2055 @@
...
+                    }
+                }
+            },
+            "put": {
+                "tags": [
+                    "EventSubscriptions"
+                ],
+                "summary": "Create an event subscription",
+                "description": "Asynchronously creates a new event subscription to the specified scope. Existing event subscriptions cannot be updated with this API and should instead use the Update event subscription API.",
+                "operationId": "EventSubscriptions_Create",
```
[the remark this answers is not in the document]
Comment (anuchandy) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json]: Similar to `Topics_CreateOrUpdate`

**Comment by anuchandy** [comment]
@ravbhatnagar new api version. Please take a look. I have looped you into the email thread with @kalyanaj

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json [hunk 3]
```diff
@@ -0,0 +1,2055 @@
...
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventGrid/eventSubscriptions": {
+            "get": {
+                "tags": [
+                    "EventSubscriptions"
+                ],
+                "summary": "List all global event subscriptions under an Azure subscription and resource group",
+                "description": "List all global event subscriptions under a specific Azure subscription and resource group",
+                "operationId": "EventSubscriptions_ListGlobalByResourceGroup",
```
[the remark this answers is not in the document]
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json]: @anuchandy, thanks for taking a look and for your comments. The service supports both regional and global event subscriptions and that's the reason we wanted to explicitly specify which ones are being listed (and not treat one category as the default implicitly). We arrived at these naming after many discussions on the PRs for the earlier two API versions, and would like to retain the same name due to the reason outlined above.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json [hunk 4]
```diff
@@ -0,0 +1,2055 @@
...
+                        "schema": {
+                            "$ref": "#/definitions/EventTypesListResult"
+                        }
+                    },
+                    "default": {
+                        "description": "*** Error Responses: ***\n\n * 400 Bad Request.\n\n * 404 Not Found.\n\n * 500 Internal Server Error."
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLi
```
[the remark this answers is not in the document]
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json]: No, we don't currently support pagination for this.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json [hunk 5]
```diff
@@ -0,0 +1,2055 @@
...
+            }
+        },
+        "/subscriptions/{subscriptionId}/providers/Microsoft.EventGrid/eventSubscriptions": {
+            "get": {
+                "tags": [
+                    "EventSubscriptions"
+                ],
+                "summary": "Get an aggregated list of all global event subscriptions under an Azure subscription",
+                "description": "List all aggregated global event subscriptions under a specific Azure subscription",
+                "operationId": "EventSubscriptions_ListGlobalBySubscription",
```
[the remark this answers is not in the document]
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json]: @anuchandy, thanks for taking a look and for your comments. The service supports both regional and global event subscriptions and that's the reason we wanted to explicitly specify which ones are being listed (and not treat one category as the default implicitly). We arrived at these naming after many discussions on the PRs for the earlier two API versions, and would like to retain the same name due to the reason outlined above.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json]: Good point on the current name not accurately reflecting the contract. Resolved now.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json (hunk 1 above)
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json]: For consistency with the convention used in EventSubscriptions_List* operations, would prefer to leave this as Topics_ListBySubscription. Please let me know if you disagree.

**Review by anuchandy — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
Looks good. Signing off!

**Pull request merged**